File: src/main/java/jp/seraphr/collection/builder/ListWrapperBuilder.java
package jp.seraphr.collection.builder;

import java.util.ArrayList;
import java.util.List;

import jp.seraphr.collection.wrapper.ListWrapper;

public class ListWrapperBuilder<_Elem> implements WrapperBuilder<List<_Elem>, List<_Elem>, _Elem, ListWrapper<_Elem>> {
    List<_Elem> mList = new ArrayList<_Elem>();

    @Override
    public void add(_Elem aElement) {
        mList.add(aElement);
    }

    @Override
    public ListWrapper<_Elem> build() {
        return new ListWrapper<_Elem>(mList);
    }

}

File: src/main/scala/jp/seraphr/collection/builder/ListWrapperBuilder.scala
package jp.seraphr.collection.builder
import jp.seraphr.collection.wrapper.ListWrapper
import java.util.ArrayList

/**
 */
class ListWrapperBuilder[_Elem] extends WrapperBuilder[_Elem, ListWrapper[_Elem]] {
  import java.util.{ List => JList }

  private val mBase = new ArrayList[_Elem]

  override def add(aElement: _Elem): Unit = mBase.add(aElement)
  override def build(): ListWrapper[_Elem] = new ListWrapper(mBase)
}